google/cloud/documentai_toolbox/converters/config/block.py

```diff
@@ -0,0 +1,239 @@
+        init=False, repr=False, default=None
+    )
+    bounding_unit: Optional[SimpleNamespace] = dataclasses.field(
+        init=False, repr=False, default=None
+    )
+    bounding_x: Optional[SimpleNamespace] = dataclasses.field(
+        init=False, repr=False, default=None
+    )
+    bounding_y: Optional[SimpleNamespace] = dataclasses.field(
+        init=False, repr=False, default=None
+    )
+    docproto_width: Optional[float] = dataclasses.field(
+        init=False, repr=False, default=None
+    )
+    docproto_height: Optional[float] = dataclasses.field(
+        init=False, repr=False, default=None
+    )
+
+    @classmethod
+    def load_blocks_from_schema(
+        cls: Type["Block"],
+        input_data: bytes,
+        input_config: bytes,
+        base_docproto: documentai.Document,
+    ) -> List["Block"]:
+        r"""Loads Blocks from original annotation data and provided config.
+
+        Args:
+            input_data (bytes):
+                Required.The bytes of the annotated data.
+            input_config (bytes):
+                Required.The bytes of config data.
+            base_docproto (bytes):
+                Required. The bytes of the original pdf.
+
+        Returns:
+            List[Block]:
+                From original annotation data and provided config.
+
+        """
+        objects = json.loads(input_data)
+        schema_json = json.loads(
+            input_config, object_hook=lambda d: SimpleNamespace(**d)
+        )
+
+        entities = schema_json.entity_object
+        type_ = schema_json.entity.type_
+
+        mention_text = schema_json.entity.mention_text
+
+        id_ = getattr(schema_json.entity, "id", None)
+        document_height = (
+            getattr(schema_json.page, "height", None)
+            if hasattr(schema_json, "page")
+            else None
+        )
+        document_width = (
+            getattr(schema_json.page, "width", None)
+            if hasattr(schema_json, "page")
+            else None
+        )
+
+        confidence = getattr(schema_json.entity, "confidence", None)
+        page_number = getattr(schema_json.entity, "page_number", None)
+        normalized_vertices = getattr(
+            schema_json.entity.normalized_vertices, "base", None
+        )
+        bounding_width = getattr(schema_json.entity.normalized_vertices, "width", None)
+        bounding_height = getattr(
+            schema_json.entity.normalized_vertices, "height", None
+        )
+        bounding_type = getattr(schema_json.entity.normalized_vertices, "type", None)
+        bounding_unit = getattr(schema_json.entity.normalized_vertices, "unit", None)
+        bounding_x = getattr(schema_json.entity.normalized_vertices, "x", None)
+        bounding_y = getattr(schema_json.entity.normalized_vertices, "y", None)
+
+        blocks: List[Block] = []
+        ens = _get_target_object(objects, entities)
+        for i in ens:
```

Comment:
The logic of this for loop and its nested if-else clauses is difficult to understand.  I get the impression that many of them could be simplified if a block `b`'s attributes have reasonable defaults.  For instance, can't

```
if bounding_width:
                b.bounding_width = _get_target_object(b.bounding_box, bounding_width)
```

be simplified to just
```
b.bounding_width = _get_target_object(b.bounding_box, bounding_width)
```
?

(Please ignore and file an internal clean up issue if this file was simply renamed.)

Reply:
This was a refactoring that I did as well. So I'm good to make some changes as needed.